docs/requests/body.md

```diff
@@ -10,25 +10,33 @@ In its simplest form, the request's body can be set as a `String`. By default, t
 
 A `String` body can be set on a request as follows:
 
-```scala
+```scala mdoc:compile-only
+import sttp.client._
 basicRequest.body("Hello, world!")
 ```
 
 It is also possible to use a different character encoding:
 
-```scala
-def body(b: String)
-def body(b: String, encoding: String)
+```scala mdoc:compile-only
+import sttp.client._
+basicRequest.body("Hello, world!", "utf-8")
 ```
 
 ## Binary data
 
 To set a binary-data body, the following methods are available:
 
-```scala
-def body(b: Array[Byte])
-def body(b: ByteBuffer)
-def body(b: InputStream)
+```scala mdoc:compile-only
+import sttp.client._
+
+val textAsBytes = "Hello, world!".getBytes
```

Comment:
I think it might be clearer to have:
```
val bytes: Array[Byte] = ???
basicRequest.body(bytes)

val byteBuffer: java.nio.ByteBuffer = ???
basicRequest.bod(byteBuffer)
```

etc. It will be clear (through type annotations) what the signatures are (as we can't show the signatures themselves)